Set stable edge IDs prior to OSM speed parsing

mark-idleman opened the pull request:
## Background
This is the first big change required to support SEID-based custom speeds in graphhopper. Conceptually, it changes when we set SEIDs for each graphhopper edge from after the main OSM import step to during the reading of OSM, before speeds are parsed. This will unlock the ability to set SEID-based custom speeds for edges during this import process (each graphhopper edge will need an SEID already set in order to apply any custom speeds passed in).

## Details
A more thorough rundown of the changes:
- During GraphHopper's import phase in `CustomOsmReader`, we make changes to ensure SEIDs are set prior to the step that parses speeds for each OSM Way (which is what we'll modify to apply SEID-based custom speeds in a subsequent PR).
  - The parsing of OSM ID, which normally occurs in the call to `osmParsers.handleWayTags()`, is pulled out and done earlier in the import process, because OSM IDs are needed to generate SEIDs. We create a new `CustomOSMParsers` class to handle the new behavior, ensuring we can do this OSM ID parsing early and not repeat it in the subsequent call to `osmParsers.handleWayTags()`
  - SEIDs are then created with new(ish) logic in `stableEdgeIdManager.setStableEdgeIdsForEdge()`/`StableIdEncodedValues`
- The above^ changes now ensure that any time the graph is imported by GraphHopper, SEIDs are applied. This allows us to peel out the function we'd created to set SEIDs post-import, `stableEdgeIdManager.setStableEdgeIds()`, which is used in `ImportCommand` (builds regular routers), `ImportSandboxCommand` (builds nationwide sandbox routers), and `ExportNationwideCommand` (which creates the street export CSV)

Note that nothing changes around how SEIDs are accessed once they've been set (`StableIdPathDetailsBuilder`, which adds SEIDs to routes output by the routing server, and `StreetEdgeExporter`, which generates the street export CSV). 
  

## Testing
- All unit tests pass ✅ 
- [Successful FT run](https://github.com/replicahq/graphhopper/actions/runs/14409973222) off of this branch (shows that returned routes remain stable vs. the "golden set")
- Built nationwide 2024_Q4 street export table [with this change in place](https://github.com/replicahq/model/compare/test_new_seid_setting_scheme?expand=1), and compared the SEIDs in the resulting table with the 2024_Q4 table built off of master. The IDs match exactly across the two versions 🎉 
<img width="821" alt="Screenshot 2025-04-11 at 11 45 24 AM" src="https://github.com/user-attachments/assets/48c80d1a-b144-4d56-b219-cba305192e36" />

mark-idleman committed:
first pass

mark-idleman committed:
remove unused imports

mark-idleman committed:
cleanup unused function

mark-idleman committed:
Merge branch 'original-direction' into set_seids_early

mark-idleman committed:
rm todo

mark-idleman committed:
rm unused functions

mark-idleman committed:
rm unused import

mark-idleman committed:
rm comment that's not yet relevant

mark-idleman commented on replica-common/src/main/java/com/graphhopper/reader/osm/CustomOsmReader.java:
```diff
@@ -368,7 +371,13 @@ else if (!config.getElevationSmoothing().isEmpty())
+        // SEIDs require OSM IDs to be calculated, so we must parse those first
+        osmParsers.handleOsmIdWayTags(edge.getEdge(), edgeIntAccess, way, relationFlags);
+        stableEdgeIdManager.setStableEdgeIdsForEdge(edge, segmentIndex);
         osmParsers.handleWayTags(edge.getEdge(), edgeIntAccess, way, relationFlags);
```
`handleWayTags` is where we'll apply user-provided SEID-based custom speeds, in a future PR

kaelgreco reviewed (approved):
awesome! glad that tests are passing. Not sure the granularity of testing but Is it worth updating any of the unit tests to capture the seid assignment changes to to `addEdge`?

mark-idleman committed:
add unit test for CustomOSMParsers

mark-idleman committed:
add test for StableEdgeIdManager

mark-idleman commented:
@kaelgreco good question! The actual `addEdge` function is pretty low-level and we don't have any replica-specific testing set up for it, but the existing testing pretty well covers end-to-end the functionality affected by the change there (ie, we still set OSM IDs + SEIDs correctly).

But, I added a few new unit tests to hit the new custom function calls we added in `addEdge`, so now we have a bit more coverage there

mark-idleman committed:
add comments to test
